nextcord/application_command.py

```diff
@@ -1243,75 +1264,183 @@ def __init__(
+            grouped_annotations: List[
+                Union[type, Annotated[Optional[OptionConverter], object], Type[None]]
+            ] = []
+            grouped_annotations.extend(unpacked_annotations)
+            grouped_annotations.extend(literals)
+            # The only literals in this should be OptionConverters. Anything else should have triggered the ValueError.
+            for anno in grouped_annotations:
+                if isinstance(anno, object) and isinstance(anno, OptionConverter):
+                    # If the annotation is instantiated, add it to the converters and set the anno to the type it has.
+                    annotation_converters.append(anno)
+                    anno = anno.type
+                elif isinstance(anno, type) and issubclass(anno, OptionConverter):
+                    # If the annotation is NOT instantiated, instantiate it and do the above.
+                    made_converter = anno()
+                    annotation_converters.append(made_converter)
+                    anno = made_converter.type
+
+                if anno is None or anno == type(None):  # type: ignore
+                    # Ignore whatever your IDE tells you about the `==`, using `is` makes it not work.
+                    # If None is included, they want it to be optional. But we don't want None processed fully as anno.
+                    annotation_required = False
+                else:
+                    if found_type is MISSING:
+                        # If we haven't set the type of the annotation, set it.
+                        found_type = self.get_type(anno)
+                    elif self.get_type(anno) is not found_type:
+                        raise ValueError(
+                            f"{self.error_name} | Annotation {anno} is incompatible with {found_type}"
+                        )
+
+                    # Annotation specific changes.
+                    if anno is CategoryChannel:
+                        found_channel_types.append(ChannelType.category)
+                    elif anno is DMChannel:
+                        found_channel_types.append(ChannelType.private)
+                    elif anno is GroupChannel:
+                        found_channel_types.append(ChannelType.group)
+                    elif anno is StageChannel:
+                        found_channel_types.append(ChannelType.stage_voice)
+                    elif anno is TextChannel:
+                        found_channel_types.append(ChannelType.text)
+                        found_channel_types.append(ChannelType.news)
+                    elif anno is VoiceChannel:
+                        found_channel_types.append(ChannelType.voice)
+                    elif anno is Thread:
+                        found_channel_types.append(ChannelType.news_thread)
+                        found_channel_types.append(ChannelType.public_thread)
+                        found_channel_types.append(ChannelType.private_thread)
```

Comment:
```suggestion
                    channel_map: dict[type[Any], tuple[ChannelType, ...]] = {
                        CategoryChannel: (ChannelType.category,),
                        DMChannel: (ChannelType.private,),
                        GroupChannel: (ChannelType.group,),
                        StageChannel: (ChannelType.stage_voice,),
                        TextChannel: (ChannelType.text, ChannelType.news),
                        VoiceChannel: (ChannelType.voice,),
                        Thread: (ChannelType.news_thread, ChannelType.public_thread, ChannelType.private_thread),
                    }

                    # Annotation specific changes.
                    if channel_types := channel_map.get(anno):
                        found_channel_types.extend(channel_types)
```
`channel_map` can (and probably should) be a top-level file constant instead, but here is the idea of https://github.com/nextcord/nextcord/pull/678/files#r912295240

Reply:
Moved to be next to option_types, implemented.